invenio_vocabularies/services/service.py

```diff
@@ -20,97 +20,64 @@
-from invenio_records_resources.services.records.params import (
-    FilterParam,
-    SuggestQueryParser,
-)
+from invenio_records_resources.services.base.utils import map_search_params
 from invenio_records_resources.services.records.schema import ServiceSchemaWrapper
 from invenio_records_resources.services.uow import unit_of_work
 from invenio_search.engine import dsl
 
-from ..records.api import Vocabulary
 from ..records.models import VocabularyType
-from .components import PIDComponent, VocabularyTypeComponent
-from .permissions import PermissionPolicy
-from .schema import TaskSchema, VocabularySchema
 from .tasks import process_datastream
 
 
-class VocabularySearchOptions(SearchOptions):
-    """Search options."""
-
-    params_interpreters_cls = [
-        FilterParam.factory(param="tags", field="tags"),
-    ] + SearchOptions.params_interpreters_cls
-
-    suggest_parser_cls = SuggestQueryParser.factory(
-        fields=[
-            "id.text^100",
-            "id.text._2gram",
-            "id.text._3gram",
-            "title.en^5",
-            "title.en._2gram",
-            "title.en._3gram",
-        ],
-    )
-
-    sort_default = "bestmatch"
-
-    sort_default_no_query = "title"
-
-    sort_options = {
-        "bestmatch": dict(
-            title=_("Best match"),
-            fields=["_score"],  # ES defaults to desc on `_score` field
-        ),
-        "title": dict(
-            title=_("Title"),
-            fields=["title_sort"],
-        ),
-        "newest": dict(
-            title=_("Newest"),
-            fields=["-created"],
-        ),
-        "oldest": dict(
-            title=_("Oldest"),
-            fields=["created"],
-        ),
-    }
-
-
-class VocabulariesServiceConfig(RecordServiceConfig):
-    """Vocabulary service configuration."""
-
-    service_id = "vocabularies"
-    indexer_queue_name = "vocabularies"
-    permission_policy_cls = PermissionPolicy
-    record_cls = Vocabulary
-    schema = VocabularySchema
-    task_schema = TaskSchema
-
-    search = VocabularySearchOptions
-
-    components = [
-        # Order of components are important!
-        VocabularyTypeComponent,
-        DataComponent,
-        PIDComponent,
-    ]
-
-    links_item = {
-        "self": Link(
-            "{+api}/vocabularies/{type}/{id}",
-            vars=lambda record, vars: vars.update(
-                {
-                    "id": record.pid.pid_value,
-                    "type": record.type.id,
-                }
-            ),
-        ),
-    }
+class VocabularyTypeService(RecordService):
+    """Vocabulary type service."""
+
+    @property
+    def schema(self):
+        """Returns the data schema instance."""
+        return ServiceSchemaWrapper(self, schema=self.config.schema)
+
+    @property
+    def links_item_tpl(self):
+        """Item links template."""
+        return LinksTemplate(
+            self.config.vocabularies_listing_item,
+        )
```

Comment:
```suggestion
```

With some of the changes suggested in the config, these will not be needed anymore.

Reply:
The view does not render without the schema property present.